maskit/ensembles.py

```diff
@@ -0,0 +1,179 @@
+from collections import deque
+from typing import Dict, NamedTuple, Optional
+from pennylane import numpy as np
+
+from maskit.masks import MaskedCircuit
+
+
+class EnsembleResult(NamedTuple):
```

Comment:
Consider to use sphinx-style attribute docstrings: a leading ``#:`` comment before the attribute. For example:
```python3
    #: branch that performs best
    branch: MaskedCircuit
```
Caveat: No matter what you do, Sphinx doesn't properly document attributes on NamedTuples last time I checked. Once it does, this is the correct style.